docs/how_to_guides/miscellaneous/how_to_add_and_test_a_new_sqlalchemydataset_class.rst

```diff
@@ -0,0 +1,215 @@
+
+The core team will not be able to merge your contribution until they're able to replicate your setup.
+
+
+2. Make changes to sqlalchemy_dataset.py
+########################################
+
+    * **At the top the file, add a try-except block**
+
+        .. code-block:: python
+
+            try:
+                import pyhive.sqlalchemy_presto as presto
+            except ImportError:
+                presto = None
+
+
+    * **In SqlAlchemyDataset.__init__, add an elif clause**
+
+        .. code-block:: python
+
+            elif self.engine.dialect.name.lower() == "presto":
+                self.dialect = import_module("pyhive.sqlalchemy_presto")
+
+    * **In SqlAlchemyDataset.create_temporary_table, add an elif clause**
+
+        .. code-block:: python
+
+            elif (self.engine.dialect.name.lower() == "awsathena"
+              or self.engine.dialect.name.lower() == "presto"):
+                stmt = "CREATE OR REPLACE TABLE {table_name} AS {custom_sql}".format(
+                    table_name=table_name, custom_sql=custom_sql)
+
+
+3. Make changes to conftest.py
+##############################
+
+    * **In pytest_addoption, add an option**
+
+        .. code-block:: python
+
+            parser.addoption(
+                "--no-presto",
+                action="store_true",
+                help="If set, suppress tests against presto",
+            )
+
+    * **In build_test_backends_list, add a variable and if clause**
+
+        .. code-block:: python
+
+            no_presto = metafunc.config.getoption("--no-presto")
+            if not no_presto:
+                presto_conn_str = "presto://presto@localhost/memory/test_ci"
+                try:
+                    engine = sa.create_engine(presto_conn_str)
+                    conn = engine.connect()
+                except (ImportError, sa.exc.SQLAlchemyError):
+                    raise ImportError(
+                        "presto tests are requested, but unable to connect to the presto database at "
+                        f"'{presto_conn_str}'"
+                    )
+                test_backends += ["presto"]
+
+    * **In the sa fixture method, add your test_backend to the list of backends**
+
+        .. code-block:: python
+
+            if "postgresql" not in test_backends and "sqlite" not in test_backends and "presto" not in test_backends:
+
+
+    * **For each of the test datasets, add a schema entry for the dialect.**
+
+        Examples:
+
+            * In ``numeric_high_card_dataset`` : ``"presto": {"norm_0_1": "DOUBLE"},``
+            * In ``datetime_dataset`` : ``"presto": {"datetime": "TIMESTAMP"},``
+            * In ``dataset_sample_data`` : ``"presto": {"infinities": "DOUBLE", "nulls": "DOUBLE", "naturals": "DOUBLE"},``
+
+
+4. Make changes to tests/test_utils.py
+######################################
+
+    * **Add a try-except clause to import dialect-specific types and map them to generic types.**
```

Comment:
Those are *not* supposed to be generic types; they should correspond to the dialect-specific correct type names that will be used in test schema definitions.

Reply:
Ah, that makes more sense.

Reply:
```suggestion
    * **Add a try-except clause to import dialect-specific types and map them to type names that will be used in test schema definitions.**
```